### PR TITLE
First steps to add line numbers

### PR DESCRIPTION
**Pull request by jneira**
Those are the few changes i 've done

### CODE REVIEW

File: src/Codec/JVM/ASM/Code/Instr.hs [hunk 1]
```diff
@@ -32,7 +32,8 @@ data InstrState =
              , isLabelTable    :: LabelTable
              , isLastBranch    :: LastBranch
              , isRunAgain      :: Bool
-             , isNextLabel     :: Int }
+             , isNextLabel     :: Int 
+             , isLineNumber    :: Maybe Int }
```
Comment (rahulmutt) [inline src/Codec/JVM/ASM/Code/Instr.hs]: Shouldn't `isLineNumber` be a Map since you can have many entries for a given method body?

**Review by rahulmutt — COMMENTED** [review]
Great work! Just had a question about your choice of representation for `isLineNumber`.

File: src/Codec/JVM/ASM/Code/Instr.hs (hunk 1 above)
Comment (jneira) [reply src/Codec/JVM/ASM/Code/Instr.hs]: Mmm, i clearly have to study the lib more carefully: i mistakenly assumed Instr represented one bytecode op

Comment (rahulmutt) [reply src/Codec/JVM/ASM/Code/Instr.hs]: No problem! I guess the naming is a bit off as well. `Instr` represents a monadic action, which when executed generates bytecode and other relevant information like attributes that can almost directly be written to classfiles.

**Review by rahulmutt — APPROVED** [review]
Well done! Love the use of newtypes to make things strongly typed.

**Pull request merged**